src/colorpicker.cpp

```diff
@@ -22,26 +22,53 @@ ColorPicker::ColorPicker(Widget *parent, const Color& color) : PopupButton(paren
     Popup *popup = this->popup();
     popup->setLayout(new GroupLayout());
 
-    mColorWheel = new ColorWheel(popup);
+    // initialize callback to do nothing; this is for users to hook into
+    // receiving a new color value
+    mCallback = [](const Color &) {};
+
+    // set the color wheel to the specified color
+    mColorWheel = new ColorWheel(popup, color);
+
+    // set the pick button to the specified color
     mPickButton = new Button(popup, "Pick");
-    mPickButton->setFixedSize(Vector2i(100, 25));
+    mPickButton->setBackgroundColor(color);
+    mPickButton->setTextColor(color.contrastingColor());
+    mPickButton->setFixedSize(Vector2i(100, 20));
+
+    // set the reset button to the specified color
+    mResetButton = new Button(popup, "Reset");
+    mResetButton->setBackgroundColor(color);
+    mResetButton->setTextColor(color.contrastingColor());
+    mResetButton->setFixedSize(Vector2i(100, 20));
 
     PopupButton::setChangeCallback([&](bool) {
-        setColor(backgroundColor());
-        mCallback(backgroundColor());
+        if (this->mPickButton->pushed()) {
+            setColor(backgroundColor());
+            mCallback(backgroundColor());
```

Comment:
Callback was getting executed too frequently, e.g. when a user changes the color wheel value but does NOT select "Pick".  Remaining changes in this file relate to this, as well as the reset button.

Reply:
One option would be to split that into two callbacks. For instance, the slider widget has a fast callback and a slow one, which may be useful for quick interactive visualizations.

Reply:
I'm not sure I understand.  If you change the current `master` of NanoGUI example1 to use the `ColorPicker`, it might be clearer why this change is here.  I rewrote `ColorPicker` to match the existing example1 functionality, specifically that the button color does not change until it's been selected.  The reset button was added (can be removed) to allow the color wheel to be reset to the actively selected color.

How does a different callback come in?

Reply:
Maybe it's me who is misunderstanding. I thought that it would be nice to have interactive feedback when e.g. changing the color of a mesh using the color picker. That requires a fast callback (i.e. "user is currently tweaking the color") and a slow one ("Pick/Reset button was finally pressed"). Potentially it will make things too complicated though.